packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -25,10 +25,21 @@ if (
   const name = member.id as TSESTree.StringLiteral;
 }
     `,
+    `
+      type Bar = 'bar';
+      const data = {
+        x: 'foo' as 'foo',
+        y: 'bar' as Bar,
+      };
+    `,
+    "[1, 2, 3, 4, 5].map(x => [x, 'A' + x] as [number, string]);",
+    `
+      let x: Array<[number, string]> = [1, 2, 3, 4, 5].map(
+        x => [x, 'A' + x] as [number, string],
+      );
+    `,
     'const foo = 3 as number;',
     'const foo = <number>3;',
-    'const foo = <3>3;',
-    'const foo = 3 as 3;',
```

Comment:
According to the original issue, only rule docs are invalid. If you believe that the current behavior of the rule is invalid, then you should open a separate issue so we can discuss this.

Reply:
Sorry about that — I didn't want to create noise by opening a second issue as I thought that issue was already raised in the first issue. However, I have done so ✅

Reply:
> I didn't want to create noise by opening a second issue

Don't worry! It's always better to open a separate issue!